crates/bevy_ui/src/focus.rs

```diff
@@ -9,9 +9,18 @@ use bevy_transform::components::GlobalTransform;
 use bevy_window::Windows;
 use smallvec::SmallVec;
 
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum ClickedWith {
```

Comment:
Instead of redefining the mouse buttons, can we just reuse the `MouseButton` enum?
```rust
pub enum ClickedWith {
    Mouse(MouseButton),
    Touch,
}
```

Reply:
I didn't do that to avoid adding an extra layer to deconstruct when matching... I would prefer not to, but ok to change if people really feel it's better to reuse and add a layer

Currently it's
```rust
Interaction::Clicked(ClickedWith::MouseLeftButton)
```

With this it would be
```rust
Interaction::Clicked(ClickedWith::Mouse(MouseButton::Left))
```
Which starts to be painful when matching

Reply:
That would only work if you want to react the same way to any mouse button click, Bevy shouldn't force that on users

Reply:
> I would prefer not to, but ok to change if people really feel it's better to reuse and add a layer

I'm personally a big fan of reusing stuff we already have. 
It's fair that it is more code to write however, it comes at the benefit of reusing existing code which is better. 

Also you could probably change `ClickedWith` to just `Clicked` to reduce a few characters :)
And you could also then do `impl From<MouseButton> for Clicked`

However at the end of the day, if most people don't want to reuse `MouseButton` I'm fine with leaving it as is. (I just have a preference for re-usability)